euphoria-core/src/main/java/cz/seznam/euphoria/core/executor/greduce/GroupReducer.java

```diff
@@ -183,6 +185,18 @@ private void onTimerCallback(long stamp, WID window) {
         newWindow = target;
       }
 
+      // ~ as of now on, we can assume `sources` do _not_ contain `target`;
+      // this implies:
+      //  a) the target window's state will not be merged into itself
+      //  b) the target window's triggers will not be merged into itself
+      //  c) the target window's trigger #onClear won't be called
+      sources.remove(target);
+
+      // XXX only if sources non empty!
```

Comment:
indeed, it's a TODO i forgot about. thanks for pointing it out! yes, skipping/continuing can/should be done. will fix this.